### PR TITLE
Adding support for removing file from git index

### PR DESCRIPTION
**Pull request by manuelfalcon**
## Hey, I just made a Pull Request!

Added support for removing file from git index. [isomorphic remove](https://isomorphic-git.org/docs/en/remove)

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by tudi2d — APPROVED** [review]
looks good to me - is there any risk with it, @benjdlambert?

File: .changeset/brave-shirts-hang.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/backend-common': minor
```
Comment (benjdlambert) [inline .changeset/brave-shirts-hang.md:2]: Think this should be `patch`

**Review by benjdlambert — CHANGES_REQUESTED** [review]
Nice thanks! Just one comment about the changeset and we're good to go.

File: .changeset/brave-shirts-hang.md (hunk 1 above)
Comment (manuelfalcon) [reply .changeset/brave-shirts-hang.md]: Agree. Done :)

**Comment by manuelfalcon** [comment]
@benjdlambert let me know if the change is looking good now :)

**Review by benjdlambert — APPROVED** [review]
Nice thanks! :Tada:

**Pull request merged**